tests_integration/nativeapp/test_large_upload.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import uuid
+import os
+from unittest import mock
+
+from snowflake.cli.api.project.util import generate_user_env
+from snowflake.connector.constants import S3_CHUNK_SIZE
+from snowflake.cli.api.secure_path import SecurePath
+from snowflake.cli.api.project.definition_manager import DefinitionManager
+from snowflake.cli.plugins.nativeapp.manager import NativeAppManager
+from snowflake.cli.plugins.nativeapp.v2_conversions.v2_to_v1_decorator import (
+    _pdf_v2_to_v1,
+)
+from snowflake.cli.plugins.stage.md5 import parse_multipart_md5sum
+
+from tests.project.fixtures import *
+from tests_integration.test_utils import pushd, enable_definition_v2_feature_flag
+
+USER_NAME = f"user_{uuid.uuid4().hex}"
+TEST_ENV = generate_user_env(USER_NAME)
+
+THRESHOLD_BYTES: int | None = None  # if set, passes this option with PUT
+TEMP_FILE_SIZE_BYTES = 200 * 1024 * 1024
+
+from contextlib import contextmanager
+
+
+@contextmanager
+def mocked_testenv():
+    """
+    Mocks all ways we can extract env vars to ensure our ctx.env is correct.
+    """
+    from os import getenv as original_getenv
+
+    def mock_getenv(key: str, default: str | None = None) -> str | None:
+        if key.lower() == "user":
+            return USER_NAME
+        return original_getenv(key, default)
+
+    with mock.patch.dict(os.environ, {"USER": USER_NAME}):  # for DefinitionManager
+        with mock.patch("os.getenv", side_effect=mock_getenv):  # for ctx.env
+            yield
+
+
+@pytest.mark.skip  # FIXME: requires AWS + python connector to support threshold=<number>
```

Comment:
```suggestion
@pytest.mark.skip(reason="Requires AWS + python connector to support threshold=<number>")
```
also didn't you mention a few days ago that the feature is supported in the connector but the test user/account has insufficient privileges?

Reply:
It's actually not supported in the connector, I thought it might be. There is an account override that overrides the connector properties we could use, but it's not necessarily the right solution. I will bring it up at Snowflake CLI office hours on Thursday.